### PR TITLE
[Android/NNFW] parse option

### PR DESCRIPTION
**Pull request by jaeyun-jung**
add new constructor and parse option string in NNFW sub-plugin.

Signed-off-by: Jaeyun <jy1210.jung@samsung.com>

### CODE REVIEW

File: ext/nnstreamer/tensor_filter/tensor_filter_nnfw.c [hunk 1]
```diff
@@ -113,6 +113,46 @@ nnfw_get_accelerator (nnfw_pdata * pdata, const char *accelerators)
   }
 }
 
+/**
+ * @brief Parse accelerator and other option.
+ */
+static void
+nnfw_parse_option (const GstTensorFilterProperties * prop, nnfw_pdata * pdata)
```
Comment (kparichay) [inline ext/nnstreamer/tensor_filter/tensor_filter_nnfw.c:120]: rename to `nnfw_parse_accl_custom`?

Comment (jaeyun-jung) [reply ext/nnstreamer/tensor_filter/tensor_filter_nnfw.c:120]: Currently only accelerator, but I am not sure whether they would request another option for NNFW.
nnfw_parse_custom_option() ?

Comment (kparichay) [reply ext/nnstreamer/tensor_filter/tensor_filter_nnfw.c:120]: nnfw_parse_custom_option() sounds good.

Comment (jaeyun-jung) [reply ext/nnstreamer/tensor_filter/tensor_filter_nnfw.c]: updated

**Review by kparichay — APPROVED** [review]
LGTM

File: ext/nnstreamer/tensor_filter/tensor_filter_nnfw.c [hunk 2]
```diff
@@ -113,6 +113,47 @@ nnfw_get_accelerator (nnfw_pdata * pdata, const char *accelerators)
...
+
+    for (i = 0; i < len; ++i) {
+      gchar **option = g_strsplit (options[i], ":", -1);
+
+      if (g_strv_length (option) > 1) {
+        g_strstrip (option[0]);
+        g_strstrip (option[1]);
+
+        if (g_ascii_strcasecmp (option[0], "Runtime") == 0) {
+          pdata->accelerator = g_strdup (option[1]);
```
Comment (dongju-chae) [inline ext/nnstreamer/tensor_filter/tensor_filter_nnfw.c:138]: Q. Is there any possibility for `pdata->accelerator` to be overwritten in this for loop?

Comment (jaeyun-jung) [reply ext/nnstreamer/tensor_filter/tensor_filter_nnfw.c:138]: currently no, nnfw_parse_custom_option() is always called with newly allocated nnfw_pdata ptr.

File: api/android/api/src/main/java/org/nnsuite/nnstreamer/SingleShot.java [hunk 3]
```diff
@@ -54,7 +54,7 @@
      * @throws IllegalStateException if this failed to construct the pipeline
      */
     public SingleShot(@NonNull File model) {
-        this(model, null, null);
+        this(new File[]{model}, null, null, NNStreamer.NNFWType.TENSORFLOW_LITE, null);
```
Comment (myungjoo) [inline api/android/api/src/main/java/org/nnsuite/nnstreamer/SingleShot.java:57]: Why TFLITE is chosen here?

Comment (jaeyun-jung) [reply api/android/api/src/main/java/org/nnsuite/nnstreamer/SingleShot.java:57]: constructor `public SingleShot(@NonNull File model)` with default tf-lite now.

**Review by dongju-chae — APPROVED** [review]
LGTM

**Comment by myungjoo** [comment]
For next releases, we will need to parse file extensions and change default engines dynamically with constructors (e.g., nnfw runtime needs to be more prioritized.) and the existence of frameworks should be checked/considered with constructor as well. (we may "cache" status of frameworks for faster constructors.)

**Pull request merged**